### PR TITLE
chore: Adopt kubernetes recommand label

### LINKED ISSUE #843: Adopt Kubernetes recommended labels
This project does not use Kubernetes [recommended labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/), e.g. `app` instead of `app.kubernetes.io/name`.

* https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml#L31
* https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml#L70

**What happened**:

Installed the snapshot-controller.

**What you expected to happen**:

The components consistently use the the recommended labels.

**How to reproduce it**:

Deploy the snapshot-controller or csi-snapshotter

**Anything else we need to know?**:

n/a

**Environment**:
- Driver version: 6.2.1
- Kubernetes version (use `kubectl version`): 1.27
- OS (e.g. from /etc/os-release): Amazon Linux 2
- Kernel (e.g. `uname -a`): 5.10
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by mowangdk**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup

**What this PR does / why we need it**:
Adopt kubernetes recommand label

**Which issue(s) this PR fixes**:

Fixes #843

**Special notes for your reviewer**:
None

**Does this PR introduce a user-facing change?**:

```release-note
Adopt Kubernetes recommended label "app.kubernetes.io/name" when deploying csi-snapshotter, snapshot-controller, and snapshot-validation-webhook.
```

### CODE REVIEW

**Comment by mowangdk** [comment]
I see that the three failed tests have the same error, but the pods are running fine after I deployed rbac-external-provisioner.yaml & setup-snapshot-controller.yaml in my local cluster. I suspect the reason  isthat the ci program or the use of old labels to get pod deployment information?

**Review by nixpanic — COMMENTED** [review]
The CI jobs fail because `release-tools/prow.sh` looks for the `app=snapshot-controller` label.

https://github.com/kubernetes-csi/external-snapshotter/blob/98b244f0d421c493d840ce7822b0a720ac8ac67f/release-tools/prow.sh#LL875C59-L875C82

**Comment by mowangdk** [comment]
Looks like I need to modify release tools? 
https://github.com/kubernetes-csi/csi-release-tools/blob/4133d1df083eaa65bdeddd0530d54278529c7a60/prow.sh#L875

**Comment by nixpanic** [comment]
> Looks like I need to modify release tools? https://github.com/kubernetes-csi/csi-release-tools/blob/4133d1df083eaa65bdeddd0530d54278529c7a60/prow.sh#L875

Yes, you may be able to read the label from the yaml as is done for `.metadata.namespace` in the line above it. Otherwise you will need to support both the old and the new label for the time being.

**Comment by mowangdk** [comment]
> > Looks like I need to modify release tools? https://github.com/kubernetes-csi/csi-release-tools/blob/4133d1df083eaa65bdeddd0530d54278529c7a60/prow.sh#L875
> 
> Yes, you may be able to read the label from the yaml as is done for `.metadata.namespace` in the line above it. Otherwise you will need to support both the old and the new label for the time being.
> 
> This probably works:
> 
> ```
> expected_label=$(kubectl apply --dry-run=client -o "jsonpath={.spec.template.metadata.labels['app']}" -f "$SNAPSHOT_CONTROLLER_YAML")
> while [ "$(kubectl get pods -n "$expected_namespace" -l "app=$expected_label" | grep 'Running' -c)" -lt "$expected_running_pods" ]; do
> ```

gotcha, thank you very much!

**Comment by nixpanic** [comment]
> > Looks like I need to modify release tools? https://github.com/kubernetes-csi/csi-release-tools/blob/4133d1df083eaa65bdeddd0530d54278529c7a60/prow.sh#L875
> 
> Yes, you may be able to read the label from the yaml as is done for `.metadata.namespace` in the line above it. Otherwise you will need to support both the old and the new label for the time being.
> 
> This probably works:
> 
> ```
> expected_label=$(kubectl apply --dry-run=client -o "jsonpath={.spec.template.metadata.labels['app']}" -f "$SNAPSHOT_CONTROLLER_YAML")
> while [ "$(kubectl get pods -n "$expected_namespace" -l "app=$expected_label" | grep 'Running' -c)" -lt "$expected_running_pods" ]; do
> ```

Sorry, that isn't exactly what you need. The `'app'` key for the label is not correct with the new labels, so you will need to tune it a little more.

**Comment by mowangdk** [comment]
> > > Looks like I need to modify release tools? https://github.com/kubernetes-csi/csi-release-tools/blob/4133d1df083eaa65bdeddd0530d54278529c7a60/prow.sh#L875
> > 
> > 
> > Yes, you may be able to read the label from the yaml as is done for `.metadata.namespace` in the line above it. Otherwise you will need to support both the old and the new label for the time being.
> > This probably works:
> > ```
> > expected_label=$(kubectl apply --dry-run=client -o "jsonpath={.spec.template.metadata.labels['app']}" -f "$SNAPSHOT_CONTROLLER_YAML")
> > while [ "$(kubectl get pods -n "$expected_namespace" -l "app=$expected_label" | grep 'Running' -c)" -lt "$expected_running_pods" ]; do
> > ```
> 
> Sorry, that isn't exactly what you need. The `'app'` key for the label is not correct with the new labels, so you will need to tune it a little more.

yeah , i noticed 😆

**Comment by mowangdk** [comment]
https://github.com/kubernetes-csi/csi-release-tools/pull/228,  PTAL

**Comment by mowangdk** [comment]
@xing-yang Please take a look to confirm that this modification meets the requirements?

**Comment by xing-yang** [comment]
@mowangdk Can you also submit a test PR that incorporates the changes from https://github.com/kubernetes-csi/csi-release-tools/pull/228?  We want to make sure it does not break existing setup.

**Comment by mowangdk** [comment]
> ncorporates the changes from [kubernetes-csi/csi-release-tools#228](https://github.com/kubernetes-csi/csi-release-tools/pull/228)
#879

**Comment by mowangdk** [comment]
@xing-yang all test passed, ptal~

File: release-tools/SIDECAR_RELEASE_PROCESS.md [hunk 1]
```diff
@@ -102,7 +102,7 @@ naming convention `<hostpath-deployment-version>-on-<kubernetes-version>`.
 1. If release was a new major/minor version, create a new `release-<minor>`
    branch at that commit.
 1. Check [image build status](https://k8s-testgrid.appspot.com/sig-storage-image-build).
-1. Promote images from k8s-staging-sig-storage to k8s.gcr.io/sig-storage. From
+1. Promote images from k8s-staging-sig-storage to registry.k8s.io/sig-storage. From
```
Comment (xing-yang) [inline release-tools/SIDECAR_RELEASE_PROCESS.md:105]: Changes in release-tools should be pulled by using subtree in a different PR.  It should not be part of this PR.

Comment (mowangdk) [reply release-tools/SIDECAR_RELEASE_PROCESS.md:105]: get~

**Comment by mowangdk** [comment]
@xing-yang ptal~

**Comment by xing-yang** [comment]
@mowangdk Can you clean up the release notes?  We can get this merged after that.

/lgtm
/approve

/hold

**Comment by xing-yang** [comment]
https://github.com/kubernetes-csi/csi-release-tools/pull/228 is already merged.

/hold cancel
